commit 2ce4f9d8618b84994b19e0a31a56562863666623
Author: Jason Andryuk <jandryuk@gmail.com>
Date:   Wed Feb 18 22:41:04 2009 +0200

    at76c50x-usb: convert at76_debug to an unsigned int
    
    at76_debug should be an unsigned int as it used as a bit field.  In
    fact, modprobe fails when trying to set at76_debug's high bit.
    
    Signed-off-by: Jason Andryuk <jandryuk@gmail.com>
    Signed-off-by: Kalle Valo <kalle.valo@iki.fi>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/at76c50x-usb.c b/drivers/net/wireless/at76c50x-usb.c
index df3efc5..aa06b90 100644
--- a/drivers/net/wireless/at76c50x-usb.c
+++ b/drivers/net/wireless/at76c50x-usb.c
@@ -109,7 +109,7 @@
 		}							 \
 	} while (0)
 
-static int at76_debug = DBG_DEFAULTS;
+static uint at76_debug = DBG_DEFAULTS;
 
 /* Protect against concurrent firmware loading and parsing */
 static struct mutex fw_mutex;
@@ -2459,7 +2459,7 @@ static void __exit at76_mod_exit(void)
 	led_trigger_unregister_simple(ledtrig_tx);
 }
 
-module_param_named(debug, at76_debug, int, 0600);
+module_param_named(debug, at76_debug, uint, 0600);
 MODULE_PARM_DESC(debug, "Debugging level");
 
 module_init(at76_mod_init);
